File: Example/Tests/CryptoUtilsTests/CryptoUtilsTests.m
//
//  CryptoUtilsTests.m
//
//  Copyright (c) 2018 Cloudinary (http://cloudinary.com)
//
//  Permission is hereby granted, free of charge, to any person obtaining a copy
//  of this software and associated documentation files (the "Software"), to deal
//  in the Software without restriction, including without limitation the rights
//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
//  copies of the Software, and to permit persons to whom the Software is
//  furnished to do so, subject to the following conditions:
//
//  The above copyright notice and this permission notice shall be included in all
//  copies or substantial portions of the Software.
//
//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
//  SOFTWARE.
//

#import <XCTest/XCTest.h>
#import <Cloudinary/Cloudinary-Swift.h>

@interface ObjCCryptoUtilsTests : XCTestCase

@end

@implementation ObjCCryptoUtilsTests

- (void)setUp {
    [super setUp];
}

- (void)tearDown {
    [super tearDown];
}

// MARK: - SHA256
- (void)test_SHA256Base64_emptyString_shouldReturnHashedString {
    
    // Given
    NSString* initialString = @"";
    
    NSString* expectedResult = @"47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU";
    
    // When
    NSString* actualResult = [CryptoObjcHelper sha256_base64WithString:initialString];
    
    // Then
    XCTAssertNotNil(actualResult, @"Hashed string should not be nil");
    XCTAssertEqualObjects(actualResult, expectedResult, "Hashed string should be equal to expected result");
}
- (void)test_SHA256Base64_specialString_shouldReturnHashedString {
    
    // Given
    NSString* initialString = @"🧼:|}!@#$%^&*()±§`~+_=-";
    
    NSString* expectedResult = @"jXJPEpRxcbKlIzNzH1RzAsaDeDR87Ir0cENeW8b5t-g";
    
    // When
    NSString* actualResult = [CryptoObjcHelper sha256_base64WithString:initialString];
    
    // Then
    XCTAssertNotNil(actualResult, @"Hashed string should not be nil");
    XCTAssertEqualObjects(actualResult, expectedResult, "Hashed string should be equal to expected result");
}
- (void)test_SHA256Base64_string_shouldReturnHashedString {
    
    // Given
    NSString* initialString = @"cryptoString";
    
    NSString* expectedResult = @"Ncsntkv4ywCQbf4Xz3pTYrxglVm02y4_X9nmCR8uNt0";
    
    // When
    NSString* actualResult = [CryptoObjcHelper sha256_base64WithString:initialString];
    
    // Then
    XCTAssertNotNil(actualResult, @"Hashed string should not be nil");
    XCTAssertEqualObjects(actualResult, expectedResult, "Hashed string should be equal to expected result");
}


@end

File: Example/Tests/CryptoUtilsTests/CryptoUtilsTests.swift
//
//  CryptoUtilsTests.swift
//  CloudinaryTests
//
//  Created by Nitzan Jaitman on 14/11/2017.
//  Copyright © 2017 Cloudinary. All rights reserved.
//

import Foundation
import XCTest
@testable import Cloudinary

class CryptoUtilsTests: XCTestCase {
    override func setUp() {
        super.setUp()
    }
    
    override func tearDown() {
        super.tearDown()
    }
    
    // MARK: - MD5
    func testMd5() {
        let md5 = "sadnkjqndlk3j43qdaoni834j032df8j0a9sdfu03124".cld_md5()
        XCTAssertEqual("e75465c18b05f1a7e665bbafa9266ef4", md5)
    }
    
    // MARK - SHA1
    func testSha1Base8(){
        let value = "sadnkjqndlk3j43qdaoni834j032df8j0a9sdfu03124"
        let secret = "rnkjd123af43214na"
        let shar1WithSecret = value.sha1_base8(secret)
        let sha1 = value.sha1_base8(nil)
        
        XCTAssertEqual("6aab31b1ad46b21ba8b1a64d451a762395ae222a", shar1WithSecret)
        XCTAssertEqual("89f30d95fd5e83ce93247677e4a301e279a48ca2", sha1)
    }
    
    func testSha1Base64(){
        let value = "sadnkjqndlk3j43qdaoni834j032df8j0a9sdfu03124"
        let sha1 = value.sha1_base64()
        
        XCTAssertEqual("ifMNlf1eg86TJHZ35KMB4nmkjKI", sha1)
    }
    
    // MARK: - SHA256
    func test_SHA256Base64_emptyString_shouldReturnHashedString() {
        
        // Given
        let initialString  = ""
        
        let expectedResult = "47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU"
        
        // When
        let actualResult = initialString.sha256_base64()
        
        // Then
        XCTAssertNotNil(actualResult, "Hashed string should not be nil")
        XCTAssertEqual(actualResult, expectedResult, "Hashed string should be equal to expected result")
    }
    func test_SHA256Base64_specialString_shouldReturnHashedString() {
        
        // Given
        let initialString  = "🧼:|}!@#$%^&*()±§`~+_=-"
        
        let expectedResult = "jXJPEpRxcbKlIzNzH1RzAsaDeDR87Ir0cENeW8b5t-g"
        
        // When
        let actualResult = initialString.sha256_base64()
        
        // Then
        XCTAssertNotNil(actualResult, "Hashed string should not be nil")
        XCTAssertEqual(actualResult, expectedResult, "Hashed string should be equal to expected result")
    }
    func test_SHA256Base64_string_shouldReturnHashedString() {
        
        // Given
        let initialString  = "cryptoString"
        
        let expectedResult = "Ncsntkv4ywCQbf4Xz3pTYrxglVm02y4_X9nmCR8uNt0"
        
        // When
        let actualResult = initialString.sha256_base64()
        
        // Then
        XCTAssertNotNil(actualResult, "Hashed string should not be nil")
        XCTAssertEqual(actualResult, expectedResult, "Hashed string should be equal to expected result")
    }
}
